kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/NMDbusConnector.java

```diff
@@ -146,71 +150,114 @@ public synchronized List<String> getInterfaces() throws DBusException {
     }
 
-    public synchronized NetworkInterfaceStatus getInterfaceStatus(String interfaceName, NetworkService networkService,
+    public synchronized NetworkInterfaceStatus getInterfaceStatus(String interfaceName,
             CommandExecutorService commandExecutorService) throws DBusException, KuraException {
-        Device device = getDeviceByIpIface(interfaceName);
-        NMDeviceType deviceType = getDeviceType(device);
-        Properties deviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME, device.getObjectPath(),
-                Properties.class);
-
-        DBusPath ip4configPath = deviceProperties.Get(NM_DEVICE_BUS_NAME, NM_DEVICE_PROPERTY_IP4CONFIG);
-        Optional<Properties> ip4configProperties = Optional.empty();
-
-        if (!ip4configPath.getPath().equals("/")) {
-            ip4configProperties = Optional
-                    .of(this.dbusConnection.getRemoteObject(NM_BUS_NAME, ip4configPath.getPath(), Properties.class));
-        }
-
-        if (!STATUS_SUPPORTED_DEVICE_TYPES.contains(deviceType)) {
-            logger.warn("Device \"{}\" of type \"{}\" currently not supported", interfaceName, deviceType);
-            return null;
-        }
-
-        Optional<UsbNetDevice> usbNetDevice = networkService.getUsbNetDevice(interfaceName);
-        if (deviceType == NMDeviceType.NM_DEVICE_TYPE_ETHERNET) {
-            Wired wiredDevice = this.dbusConnection.getRemoteObject(NM_BUS_NAME, device.getObjectPath(), Wired.class);
-            Properties wiredDeviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME,
-                    wiredDevice.getObjectPath(), Properties.class);
-
-            DevicePropertiesWrapper ethernetPropertiesWrapper = new DevicePropertiesWrapper(deviceProperties,
-                    Optional.of(wiredDeviceProperties), NMDeviceType.NM_DEVICE_TYPE_ETHERNET);
-
-            return NMStatusConverter.buildEthernetStatus(interfaceName, ethernetPropertiesWrapper, ip4configProperties,
-                    usbNetDevice);
-        } else if (deviceType == NMDeviceType.NM_DEVICE_TYPE_LOOPBACK) {
-            DevicePropertiesWrapper loopbackPropertiesWrapper = new DevicePropertiesWrapper(deviceProperties,
-                    Optional.empty(), NMDeviceType.NM_DEVICE_TYPE_LOOPBACK);
-
-            return NMStatusConverter.buildLoopbackStatus(interfaceName, loopbackPropertiesWrapper, ip4configProperties);
-        } else if (deviceType == NMDeviceType.NM_DEVICE_TYPE_WIFI) {
-            Wireless wirelessDevice = this.dbusConnection.getRemoteObject(NM_BUS_NAME, device.getObjectPath(),
-                    Wireless.class);
-            Properties wirelessDeviceProperties = this.dbusConnection.getRemoteObject(NM_BUS_NAME,
-                    wirelessDevice.getObjectPath(), Properties.class);
+        NetworkInterfaceStatus networkInterfaceStatus = null;
+        Optional<Device> device = getDeviceByInterfaceName(interfaceName);
+        if (device.isPresent()) {
```

Comment:
This is just personal preference but you can add an early return here:

```suggestion
        if (!device.isPresent()) {
            return null;
        }
```

Personally I think it's more readable, up to you really.